test/ActiveRecordSelectTest.php

```diff
@@ -46,6 +46,20 @@ public function testListOfSimpleStrings()
         ], $book->attributes());
     }
 
+    public function testRemoveRepeatedColumnNames()
+    {
+        $relation = Author::select('name,name')->select('name, author_id');
+        $this->assertEquals('SELECT name, author_id FROM `authors`', $relation->to_sql());
+    }
+
+    public function testStarRemovesAllOtherColumnNames()
+    {
+        $relation = Author::select('name')->select('*');
+        $options = $this->getPrivateVariable($relation, 'options');
```

Comment:
Per my previous comments, let's not get in the habit of doing this kind of reflection in our tests. If you want to confirm that the query is what you want, you could try the newly-added `to_sql()`:

```php

$this->assert_sql_has("select (*) from", $relation->to_sql());

```

Reply:
Ok, removed